### ISSUE TITLE
[KomaUI Plotting] obj_ui[] does not draw on first time updated. Need to press :rho or other option.

### ISSUE THREAD

**Issue by curtcorum** [JuliaHealth/KomaMRI.jl#407]
### What happened?

obj_ui[] does not draw on first time updated: 
![obj_ui_first_update](https://github.com/JuliaHealth/KomaMRI.jl/assets/1796102/5c1bb3c5-15c3-4c22-a0df-573c94b73498)

Need to press :rho or other button option:
![push_rho](https://github.com/JuliaHealth/KomaMRI.jl/assets/1796102/cba0b55e-00d7-49ba-ad26-39f36493efbd)

### Environment

```Markdown
OS x86_64-linux-gnu
Julia 1.10.4
KomaMRIPlots 0.8.3
KomaMRIFiles 0.8.2
KomaMRI 0.8.2
KomaMRICore 0.8.3
KomaMRIBase 0.8.4
```

**Comment by curtcorum** [comment]
`obj_ui[]=brain_phantom2D();`

Has same behavior. No draw the first time.

**Comment by curtcorum** [comment]
`obj_ui[]=pelvis_phantom2D()`

Also does the same thing, so consistent. Same code or same code called?

**Comment by cncastillo** [comment]
I will need to look into this, in KomaUI.jl we define the callbacks as follows:

```julia
    on((obj) -> view_ui!(obj, w, seq_ui[], widgets_button_obj; key=:ρ, darkmode), obj_ui)

    for (widget_button, key) in zip(widgets_button_obj, fieldnames_obj)
        on((cnt) -> view_ui!(cnt, w, obj_ui[], seq_ui[], widgets_button_obj; key, darkmode), widget_button)
    end
```

So, for example, the first one means that each time `obj_ui` changes the `view_ui!` function is triggered. I believe the problem are the buttons, I fixing this locally.

**Comment by cncastillo** [comment]
After looking into it, that wasn't the problem. There are two things that I didn't realize:
- `plot_phantom_map` was changed to use `Plot` instead of the backend-agnostic `plot_koma`, and that generates this problem.
- If `view_2D=true`, the plot uses `scatter` instead of `scattergl`, which is very slow.

@pvillacorta could you fix this?

**Comment by cncastillo** [comment]
@pvillacorta Also, it seems that plotting a phantom with too many spins shows fewer spins, with no warning telling the user that it is happening.

**Comment by curtcorum** [comment]
@pvillacorta @cncastillo 

It does seem to do the warning the first time only, which is probably related to the calling issues mentioned in: https://github.com/JuliaHealth/KomaMRI.jl/issues/407#issuecomment-2171855179

```
julia> obj_ui[]=brain_phantom3D(;ss=2)
┌ Warning: For performance reasons, the number of displayed spins was capped to `max_spins`=100000.
└ @ KomaMRIPlots ~/src/KomaMRI/KomaMRIPlots/src/ui/DisplayFunctions.jl:1080

```

**Comment by pvillacorta** [comment]
See `maxlog=1` in line 1080 of https://github.com/JuliaHealth/KomaMRI.jl/pull/387/commits/592b85f345fa7e3701f436ef985b25a5d2043eaf. That is why the message only displays once

**Comment by pvillacorta** [comment]
> - `plot_phantom_map` was changed to use `Plot` instead of the backend-agnostic `plot_koma`, and that generates this problem.

The solution is not as easy as using `plot_koma` again, since `PlotlyJS.plot` (which is called by `plot_koma` when using the Plotly backend) does not support frames for animations. It looks like a bug from PlotlyJS. See: 
- https://discourse.julialang.org/t/plotlyjs-jl-animation/70058/9
- https://community.plotly.com/t/adding-an-animation-button-to-a-julia-plot-with-slider/73404/2


----
> - If `view_2D=true`, the plot uses `scatter` instead of `scattergl`, which is **very slow**.

Solved in #413, as well as the `maxlog` issue

**Comment by pvillacorta** [comment]
https://github.com/JuliaHealth/KomaMRI.jl/pull/413#issuecomment-2189778867
